pkg/blobinfocache/boltdb/boltdb.go

```diff
@@ -282,12 +282,12 @@ func (bdc *cache) RecordKnownLocation(transport types.ImageTransport, scope type
 	}) // FIXME? Log error (but throttle the log volume on repeated accesses)?
 }
 
-// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket with corresponding compression info from compressionBucket (if compressionBucket is not nil), and returns the result of appending them to candidates.
+// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket with corresponding compression info from compressionBucket (if compressionBucket is not nil), and returns the result of appending them to candidates. If scope is nil, final result also contains blobs with lowest priority without locations that could be possibly be reused.
```

Comment:
Something like
```suggestion
// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket
// (or with an unknown location if scopeBucket is nil) with corresponding compression info from compressionBucket
// (if compressionBucket is not nil), and returns the result of appending them to candidates.
```
The priority of unknown locations is policy this function doesn’t decide.

Similarly for the memory backend.